### PR TITLE
Add support for Java 17 and update ASM to version 9.2

### PR DESCRIPTION
**Pull request by Stephan202**
See:
- https://asm.ow2.io/versions.html
- https://gitlab.ow2.org/asm/asm/-/compare/ASM_9_0...ASM_9_2

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -64,6 +64,7 @@
       <equals arg1="${-cleaned.specification.version}" arg2="14"/>
       <equals arg1="${-cleaned.specification.version}" arg2="15"/>
       <equals arg1="${-cleaned.specification.version}" arg2="16"/>
+      <equals arg1="${-cleaned.specification.version}" arg2="17"/>
```
Comment (Stephan202) [inline build.xml:67]: I tried to follow ac754c4bbc5e3e2a988dd266845e8ff3622781b1 as closely as possible.

File: src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-deprecated-17.txt [hunk 2]
```diff
@@ -0,0 +1,632 @@
+# This file contains API signatures extracted from the rt.jar / jimage file shipped with the class library of Oracle's Java Runtime Environment.
+# It is provided here for reference, but can easily regenerated by executing from the source folder of forbidden-apis:
+# $ ant generate-deprecated
+
+# This file contains all public, deprecated API signatures in Java version 17 (extracted from build 17-ea).
```
Comment (Stephan202) [inline src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-deprecated-17.txt:5]: Downloaded the [OpenJDK early-access build](https://jdk.java.net/17/) today. Version details:
```
$ java -version
openjdk version "17-ea" 2021-09-14
OpenJDK Runtime Environment (build 17-ea+31-2664)
OpenJDK 64-Bit Server VM (build 17-ea+31-2664, mixed mode, sharing)
```

File: src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-internal-17.txt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# This file contains API signatures that are marked as internal in Java.
+# It is provided here for reference, but can easily regenerated by executing from the source folder of forbidden-apis:
+# $ ant generate-internal
+
+# This file contains all internal packages listed in Security.getProperty("package.access") of Java version 17 (extracted from build 17-ea).
```
Comment (Stephan202) [inline src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-internal-17.txt:5]: ^ Used the same version as mentioned above.

File: src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-unsafe-17.txt [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# These methods and classes should not be used by server classes (unsafe, no charset, no locale,...):
+
+# We only include 16 for now.
+# TODO: Scan Java 17 API for additional locale/charset/... violations!
+@includeBundled jdk-unsafe-16
```
Comment (Stephan202) [inline src/main/resources/de/thetaphi/forbiddenapis/signatures/jdk-unsafe-17.txt:19]: Copied the `jdk-unsafe-16.txt` file and modified it accordingly.

**Review by Stephan202 — COMMENTED** [review]
I realize that JDK 17 isn't out yet, but hopefully this will allow us to hit the ground running once it does :)

**Comment by uschindler** [comment]
Looks fine as a start!

**Comment by uschindler** [comment]
I will update the signatures files before release, but it is unlikely that something changes.

**Pull request merged**